app/src/main/java/com/github/fribourgsdp/radio/MusixmatchLyricsGetter.kt

```diff
@@ -32,147 +30,173 @@ interface LyricsGetter{
-         * The mention to Musixmatch will be displayed elsewhere, like in the activity displaying the lyrics.
-         */
-        private fun cleanLyrics(lyrics : String) : String{
-            if(lyrics == LYRICS_NOT_FOUND) {
-                return lyrics
-            }
-            val allLines = lyrics.split("\n")
-            val onlyLyricsLines = allLines.subList(0, allLines.size-4)
-            val sj = StringJoiner("\n")
-            onlyLyricsLines.forEach { e -> sj.add(e) }
-            return sj.toString()
+    /**
+     * Transforms the lyrics by removing its 4 last lines. Indeed, the lyrics we get from Musixmatch contain a useless mention "These lyrics are not for commercial use, ..." .
+     * The mention to Musixmatch will be displayed elsewhere, like in the activity displaying the lyrics.
+     */
+    private fun cleanLyrics(lyrics : String) : String{
+        if(lyrics == LYRICS_NOT_FOUND || lyrics == "") {
+            return lyrics
         }
+        val allLines = lyrics.split("\n")
+        val onlyLyricsLines = allLines.subList(0, allLines.size-4)
+        val sj = StringJoiner("\n")
+        onlyLyricsLines.forEach { e -> sj.add(e) }
+        return sj.toString()
+    }
+
+    /**
+     * Translates lyrics to HTML format crossing out the name of the song.
+     * @param lyrics The lyrics to transform
+     * @param name The name of the song to cross
+     */
+    fun markSongName(lyrics : String, name : String) : String{
+        return if (name.isEmpty()){
+            lyrics
+        } else
+            lyrics
+                .replace(name, "<strike>${name[0].uppercase() + name.lowercase().substring(1)}</strike>", ignoreCase = true)
+                .replace("\n", "<br>")
+    }
 
-        /**
-         * Translates lyrics to HTML format crossing out the name of the song.
-         * @param lyrics The lyrics to transform
-         * @param name The name of the song to cross
-         */
-        fun markSongName(lyrics : String, name : String) : String{
-            return if (name.isEmpty()){
-                lyrics
-            } else
-                lyrics
-                    .replace(name, "<strike>${name[0].uppercase() + name.lowercase().substring(1)}</strike>", ignoreCase = true)
-                    .replace("\n", "<br>")
+    /**
+     * If status is 404, it means this song will never have lyrics.
+     * If status is not 404 but still of the form 4xx or 5xx, this request could have encountered an error, and another request could work
+     * Else if no problem execute function.
+     */
+    private fun <T> checkStatus(
+        status: Int,
+        future: CompletableFuture<T>,
+        function: () -> Boolean, ) {
+        when {
+            status == 404 -> {
+                future.completeExceptionally(NoLyricsFoundForThisSong())
+            }
+            status >= 400 -> {
```

Comment:
Couldn't this condition be > 400 instead? Since the case where status == 400 is dealt with above.

Reply:
It is the error 404 that is dealt above ;) this line is to deal with 400 (Bad request, unlikely to happen), 401 (Unauthorized, i.e. when we exhaust our API calls) and 5xx (Server errors that might always happen).
NB: 404 is handled separately because it is a response that occurs on a normal request and a normal server behavior when there are no lyrics for this song. All other errors >=400 are unexpected behaviors.